automation_tools/__init__.py

```diff
@@ -1465,3 +1469,178 @@ def setenforce(mode):
         ))
 
     run('setenforce {0}'.format(mode))
+
+
+def get_hostname_from_ip(ip):
+    """Retrives the hostname by logging into remote machine by IP.
+    Specially for the systems who doesnt support reverse DNS.
+    e.g usersys machines.
+
+    :param str ip: Remote host's IP address to retrive hostname.
+    """
+    gethost = lambda: run('hostname')
+    result = execute(gethost, host=ip)
+    print('host', result[ip])
+    return result[ip]
+
+
+def create_openstack_instance(
+        instance_name=None, image_name=None, flavor_name=None, ssh_key=None):
+    """Creates openstack Instance from Image and Assigns a floating IP \
+    to instance.
+
+    :param str instance_name: Openstack Instance name to create.
+    :param str image_name: Openstack image name from which \
+    instance to be created.
+    :param str flavor_name: Openstack flavor_name for instance. e.g m1.small.
+    :param str ssh_key: ssh_key 'name' that you want to add to this instance.
+
+    ssh_key should be added to openstack project before running automation.
+    Else the automation will fail.
+    """
+    if (instance_name or image_name or flavor_name or ssh_key) is None:
+        raise ValueError(
+            'instance_name/image_name/flavor_name/ssh_key cannot be None.')
+    network_name = 'satellite-jenkins'
+    clientObj = client.Client(
+        username='satellite-jenkins',
+        api_key='dog8code',
+        auth_url='http://qeos.lab.eng.rdu2.redhat.com:5000/v2.0',
+        project_id='satellite-jenkins'
+    )
+    # Validate openstack credentials.
+    clientObj.authenticate()
+    # Validate ssh_key is added into openstack project
+    clientObj.keypairs.find(name=ssh_key)
+    image = clientObj.images.find(name=image_name)
+    flavor = clientObj.flavors.find(name=flavor_name)
+    network = clientObj.networks.find(label=network_name)
+    floating_ip = clientObj.floating_ips.create(
+        clientObj.floating_ip_pools.list()[0].name
+    )
+    # Create instance from the given parameters
+    instance = clientObj.servers.create(
+        name=instance_name,
+        image=image.id,
+        flavor=flavor.id,
+        key_name=ssh_key,
+        network=network.id
+    )
+    # Average time considered to wait for instance to get in \
+    # ACTIVE running state, for assigning floating ip
+    time.sleep(300)
+    instance.add_floating_ip(floating_ip)
+    # Wait till DNS resolves the IP
+    time.sleep(600)
+    # Getting Hostname from IP
+    env['INSTANCE_HOST'] = get_hostname_from_ip(str(floating_ip.ip))
+
+
+def delete_openstack_instance(instance_name=None):
+    """Deletes openstack Instance.
+
+    :param str instance_name: Openstack instance name to delete.
+    """
+    if instance_name is None:
+        raise ValueError('Instance name cannot be None.')
+    clientObj = client.Client(
```

Comment:
We are duplicating the novaclient connection code, it really worth to create a function that do the all required thing and return the brand new client.